nova/scheduler/mixins.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2021 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import time
+
+from nova import context
+from nova.scheduler import client
+
+
+class HypervisorSizeFilterMixin(object):
+
+    _HV_SIZE_CACHE_RETENTION_TIME = 10 * 60
+
+    def __init__(self):
+        self._HV_SIZE_CACHE = {}
```

Comment:
Wouldn't it make sense to put the cache on the class level?
That way the cache is shared between the different filters, and as I understand from our conversation yesterday, the filters get recreated for each filtering, resulting in an empty cache.

Reply:
They don't get recreated and I even mention putting this from the class level into the instance level in the commit-message, because previously we had concurrent writers to the same dict and couldn't have different _HV_SIZE_CACHE_RETENTION_TIME settings per filter.

Reply:
https://github.com/sapcc/nova/blob/33b38a5557e9e11af2f17b6e969b255aa74c168c/nova/scheduler/host_manager.py#L489 This is where they get instantiated.

Reply:
I see.

But `_HV_SIZE_CACHE_RETENTION_TIME` is the same value in each filter/weigher, and I am unclear about the need to have different values here.

I would consider handling concurrent access inside a python application a smaller problem compared to having to make multiple calls to a Rest api for fetching essentially the same value.

On top of that, encapsulating that in one place gives us the opportunity to ensure that the values are the same across a single scheduling call.

Reply:
> On top of that, encapsulating that in one place gives us the opportunity to ensure that the values are the same across a single scheduling call.

I disagree. Retention time might kick in at any place this is used. The value should be basically static though ...

I'll switch it back to being a single dict for everyone.